R/data.table.R

```diff
@@ -2629,7 +2629,14 @@ setnames = function(x,old,new,skip_absent=FALSE) {
     #  if (anyDuplicated(new)) warningf("Some duplicates exist in 'new': ", brackify(new[duplicated(new)]))  # dups allowed without warning; warn if and when the dup causes an ambiguity
     if (anyNA(new)) stopf("NA in 'new' at positions %s", brackify(which(is.na(new))))
     if (anyDuplicated(old)) stopf("Some duplicates exist in 'old': %s", brackify(old[duplicated(old)]))
-    if (is.numeric(old)) i = old = seq_along(x)[old]  # leave it to standard R to manipulate bounds and negative numbers
+    if (is.numeric(old)) {
+      # leave it to standard R to manipulate bounds and negative numbers
+      i = old = seq_along(x)[old]
+      # For numeric old, rep_len new to be the same size as old and warn if modulo isn't 0 to match names(x)[old]<-new, #5853
+      if (length(old) %% length(new) != 0L)
+        warningf("Number of items to replace, %d, is not a multiple of replacement length, %d.", length(old), length(new))
```

Comment:
I would drop all the commas and replace them with brackets since this is the common style we use

"Number of items to replace (%d) is not a multiple of replacement length (%d).